disnake/ext/commands/base_core.py

```diff
@@ -183,6 +186,43 @@ def __init__(self, func: CommandCallback, *, name: str = None, **kwargs):
         self._before_invoke: Optional[Hook] = None
         self._after_invoke: Optional[Hook] = None
 
+    def _ensure_assignment_on_copy(self, other: AppCommandT) -> AppCommandT:
+        other._before_invoke = self._before_invoke
+        other._after_invoke = self._after_invoke
+        if self.checks != other.checks:
+            other.checks = self.checks.copy()
+        if self._buckets.valid and not other._buckets.valid:
+            other._buckets = self._buckets.copy()
+        if self._max_concurrency != other._max_concurrency:
+            # _max_concurrency won't be None at this point
+            other._max_concurrency = cast(MaxConcurrency, self._max_concurrency).copy()
+        if self.permissions != other.permissions:
+            # TODO: Maybe update instead?
+            other.permissions = self.permissions.copy()
+        if self.guild_ids and self.guild_ids != other.guild_ids:
+            other.guild_ids = self.guild_ids.copy()
```

Comment:
```suggestion
        if self.guild_ids != other.guild_ids:
            other.guild_ids = self.guild_ids.copy() if self.guild_ids is not None else None
```
Otherwise, `guild_ids = None` wouldn't be copied if it had been updated after the original `InvokableApplicationCommand` object had been created (for whatever reason)